api/v1/controllers/RSVPController.js

```diff
@@ -0,0 +1,146 @@
+}
+
+function _removeFromList(rsvpCurrent, rsvpNew) {
+    if(rsvpCurrent.get('attendeeAttendance') !== 'NO' && rsvpNew.attendeeAttendance === 'NO')
+        return true;
+    return false;
+}
+
+function _addToList(rsvpCurrent, rsvpNew) {
+    if(rsvpCurrent.get('attendeeAttendance') === 'NO' && rsvpNew.attendeeAttendance !== 'NO')
+        return true;
+    return false;
+}
+
+function createRSVP(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .createRSVP(attendee, req.user, req.body)
+                .then(function (rsvp) {
+                    if(rsvp.get('attendeeAttendance') !== 'NO')
+                        services.MailService.addToList(req.user, mail.lists.attendees);
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        });
+}
+
+function fetchRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function fetchRSVPByAttendeeId(req, res, next) {
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function updateRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function (rsvp) {
+                    if(_addToList(rsvp, req.body))
+                        services.MailService.addToList(req.user, mail.lists.attendees);
+                    if(_removeFromList(rsvp, req.body))
+                        services.MailService.removeFromList(req.user, mail.lists.attendees);
+
+                    return services.RSVPService.updateRSVP(rsvp, req.body);
+                })
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function (error) {
+            return next(error);
+        });
+}
+
+function updateRSVPById(req, res, next) {
+    services.RegistrationService
```

Comment:
Kind of the same comment as above. This looks a lot like the other update handler. Any way to add a helper function (or functions) to reduce the repeated code?